### ISSUE TITLE
Add all Bootstrap 4 event properties to custom events (in addition to relatedTarget)

### ISSUE THREAD

**Issue by picasticks** [thednp/bootstrap.native#394]
I've got a pull request ready that I'll send after submitting this issue.

I'm using the Carousel component and was using the "direction", "from" and "to" properties on the slide event. These [are described here in the Bootstrap 4 docs](https://getbootstrap.com/docs/4.0/components/carousel/#events). I didn't see these in bootstrap.native so I decided to add it.

Here's what I've done so far in the PR:

1. Modified util/bootstrapCustomEvent so that instead of taking relatedTarget as the third function argument, it instead accepts a list of properties that are then copied onto the event. This is similar to how Bootstrap's custom event handler works.

2. Updated carousel to use this new syntax for the slide and slid events and to pass `direction`, `from`, `to` as well as `relatedTarget`. I've built and tested everything so far in Firefox and Chromium. Here you can see all the properties in browser console:
![image](https://user-images.githubusercontent.com/3988667/98756598-e93ca900-2398-11eb-87c8-9afcf8519408.png)

3. I updated all the other components (dropdown, modal, scrollspy, tab) that were passing 3 arguments to `bootstrapCustomEvent()` so that they work with the new modified `bootstrapCustomEvent()` syntax. However since I don't use these components I have not regression tested anything. I also haven't looked at the Bootstrap 4 docs to see if any other components besides Carousel add useful properties to the custom events that we would also want to add in bootstrap.native. (I could maybe work on that, but wanted to submit this first to see what you think)

Thanks for considering the PR. Of course if you prefer to implement this in a different way that would be great too.

**Comment by thednp** [comment]
Thanks for the PR. Much appreciated.

Please consider:
* updating all components accordingly into your fork
* updating the `index.html` in the `Events` of each component, especially **Carousel**, a quick draft would suffice for now; users might want to know how make use of the new functionality
* any other suggestion on the functionality upgrade, since our implementation dates back to initial original jQuery plugins, find any opportunities for improvement

Thanks again.

**Comment by picasticks** [comment]
Sure, sounds like a good plan. I'll make the `index.html` updates for Carousel and then have a look at updating all the other components with any other event properties in my fork.

**Comment by picasticks** [comment]
OK, so that did not take very long at all, because there was not much else to do. For each of the 11 components I looked at the Bootstrap 4 versions and it turns out Carousel is the only one that adds any custom properties to events (other than `relatedTarget`). So no additional code changes were necessary.

Then for the 5 bootstrap.native components that I modified in my PR, I clicked through all the examples in index.html and verified in the Firefox console that when the events are triggered `event.relatedTarget` continues to be set to the correct DOM element. So that's one successful regression test.

I also click-tested the other 6 components (that I didn't modify) to verify functionality and see that they weren't throwing any errors into the console. (alert, button, collapse, popover, toast, tooltip)

I updated `index.html` with documentation for the new Carousel event properties and with a code example. That should show up in the PR.

I'll attach my build here just in case you want to verify your build is the same as what I tested.

[bootstrap-native.js.gz](https://github.com/thednp/bootstrap.native/files/5527518/bootstrap-native.js.gz)
